src/components/MultiHopTrade/components/LimitOrderV2/LimitOrderConfirm.tsx

```diff
@@ -0,0 +1,405 @@
+          .times(feeAssetRateUserCurrency)
+          .toFixed()
+        return (
+          <Stack spacing={4} width='full'>
+            <Row>
+              <Row.Label>
+                <Text translation='limitOrder.networkFee' />
+              </Row.Label>
+              <Row.Value>
+                <Skeleton isLoaded={!isLoadingAllowanceReset}>
+                  <HStack justifyContent='flex-end'>
+                    <Amount.Crypto
+                      value={allowanceResetNetworkFeeCryptoPrecision}
+                      symbol={feeAsset?.symbol ?? ''}
+                    />
+                    <Amount.Fiat
+                      color={'text.subtle'}
+                      prefix='('
+                      suffix=')'
+                      noSpace
+                      value={allowanceResetNetworkFeeUserCurrency}
+                    />
+                  </HStack>
+                </Skeleton>
+              </Row.Value>
+            </Row>
+          </Stack>
+        )
+      case LimitOrderSubmissionState.AwaitingLimitOrderSubmission:
+        return <LimitOrderDetail />
+      default:
+        return null
+    }
+  }, [
+    allowanceResetNetworkFeeCryptoBaseUnit,
+    approvalNetworkFeeCryptoBaseUnit,
+    feeAsset?.precision,
+    feeAsset?.symbol,
+    feeAssetRateUserCurrency,
+    isLoadingAllowanceApproval,
+    isLoadingAllowanceReset,
+    orderSubmissionState,
+  ])
+
+  const buttonTranslation: string | [string, number | InterpolationOptions] | undefined =
+    useMemo(() => {
+      if (!isConnected || isDemoWallet) return 'common.connectWallet'
+      switch (orderSubmissionState) {
+        case LimitOrderSubmissionState.AwaitingAllowanceApproval:
+          return ['trade.approveAsset', { symbol: sellAsset?.symbol }]
+        case LimitOrderSubmissionState.AwaitingAllowanceReset:
+          return 'trade.resetAllowance'
+        case LimitOrderSubmissionState.AwaitingLimitOrderSubmission:
+          return 'limitOrder.placeOrder'
+        default:
+          return undefined
+      }
+    }, [isConnected, isDemoWallet, orderSubmissionState, sellAsset?.symbol])
+
+  const handleConfirm = useCallback(async () => {
+    if (!isConnected || isDemoWallet) {
+      walletDispatch({ type: WalletActions.SET_WALLET_MODAL, payload: true })
+      return
+    }
+    if (!quoteId) {
+      console.error('Attempting to confirm with undefined quoteId')
+      return
+    }
+    switch (orderSubmissionState) {
+      case LimitOrderSubmissionState.AwaitingAllowanceApproval:
+        allowanceApprovalMutation.mutate()
+        break
+      case LimitOrderSubmissionState.AwaitingAllowanceReset:
+        allowanceResetMutation.mutate()
+        break
+      case LimitOrderSubmissionState.AwaitingLimitOrderSubmission:
+        const result = await placeLimitOrder({ quoteId, wallet })
+
+        // Exit if the request failed.
+        if ((result as { error: unknown }).error || !!result) {
+          setLimitOrderTxFailed(quoteId)
```

Comment:
Confirming this *does* get called but with nothing being surfaced visually:

<img width="1065" alt="image" src="https://github.com/user-attachments/assets/c905f477-c4a8-485a-baa8-923386e61640" />

Could we be dealing with wrong `quoteId` refs?

Reply:
Confirmed this is now happy when sad! 
https://jam.dev/c/68bb8ab3-0cfe-4e17-be8a-c7cddddccb7d